File: web-bff/src/modules/clients/orderProductsService/orderProductsRabbitMQ.ts
import amqp from 'amqplib';
import { auditLogsMsQueueName, orderProductsMsQueueBaseUrl, orderProductsMsQueueName } from '../../config';

let channel: amqp.Channel;
const queueName = orderProductsMsQueueName || "orders"
const logsQueueName = auditLogsMsQueueName || "audit_logs"

export const connectOrderProductsServiceMQ = async () => {
  const connection = await amqp.connect('amqp://'+orderProductsMsQueueBaseUrl);
  channel = await connection.createChannel();
  await channel.assertQueue(queueName, { durable: false });
  await channel.assertQueue(logsQueueName, { durable: true });
};

export const publishToQueueOrderProductsService = (message: object) => {
  if (!channel) throw new Error('RabbitMQ channel is not initialized.');
  const buffer = Buffer.from(JSON.stringify(message));
  channel.sendToQueue(queueName, buffer, { persistent: true });
};

export const publishToQueueAuditLogs = (text: string) => {
  console.log(text);
  if (!channel) throw new Error('RabbitMQ channel is not initialized.');
  const buffer = Buffer.from(JSON.stringify({service: "web-bff", message: text}));
  channel.sendToQueue(logsQueueName, buffer, { persistent: true });
};

File: mobile-bff/src/modules/clients/orderProductsService/orderProductsRabbitMQ.py
import pika
import json
from modules.config import ORDER_PRODUCTS_MSQUEUE_BASE_URL, ORDER_PRODUCTS_MSQUEUE_NAME

# Initialize the channel
channel = None
queue_name = ORDER_PRODUCTS_MSQUEUE_NAME or "orders"

def connect_order_products_service_mq():
    global channel
    connection = pika.BlockingConnection(pika.ConnectionParameters(host=ORDER_PRODUCTS_MSQUEUE_BASE_URL))
    channel = connection.channel()
    channel.queue_declare(queue=queue_name, durable=False)

def publish_to_queue_order_products_service(message: dict):
    if channel is None:
        raise Exception('RabbitMQ channel is not initialized.')

    channel.basic_publish(
        exchange='',
        routing_key=queue_name,
        body=json.dumps(message),
        properties=pika.BasicProperties(
            delivery_mode=2,  # Persistent message
        )
    )
